### ISSUE TITLE
Dynamically adding panels in child component throws `Previous layout not found for panel index`

### ISSUE THREAD

**Issue by davidfant** [bvaughn/react-resizable-panels#372]
I'm running into problems with defining panels within a child component and dynamically adding children. When trying to resize, the following error is console logged and the panel sizes don't change:
`Previous layout not found for panel index`

Example:
```tsx
const ChildWithDynamicPanels = () => {
  const [extra, setExtra] = useState(2);
  return (
    <>
      {Array.from({ length: extra }).map((_, i) => (
        <Fragment key={i}>
          <PanelResizeHandle />
          <Panel id={`extra-${i}`} order={i}>
            <div style={{ width: "100%", background: "red" }}>{i}</div>
          </Panel>
        </Fragment>
      ))}
      <button onClick={() => setExtra((e) => e + 1)}>Add</button>
    </>
  );
};

export function Parent({
  panels,
  children,
}: ResizablePanelsProps) {
  return (
    <PanelGroup      direction="horizontal"    >
      <ChildWithDynamicPanels />
    </PanelGroup>
  );
}
```

Behavior I'm seeing in the reproable example above:
1. Dragging the first two children works because they were defined on the initial render
2. When adding a new child, the first drag creates lots of errors and resets all sizes to be equal. Subsequent resizing works.
3. If everything in `ChildWithDynamicPanels` is moved into `Parent`, then there's no error. However, when adding a new panel, the resizing is reset and each panel gets the same size (unlike point 2 where the reset doesn't happen until dragging is attempted)

The reason I need this is because I'm using NextJS App Router, where depending on the current route additional panels might be rendered based on the route. This means that these Panels will be within a separate `page.tsx` file and thus not rendered as a direct child to the panel group. If there's a better way to do this for the NextJS App Router scenario, please let me know!

**Comment by bvaughn** [comment]
@davidfant Can you please share a runnable repro of the problem you're reporting?

The code snippet you shared doesn't really make sense. (The props passed to `Parent` aren't even use.) But more than that, it's better if I can just click a link and start looking at the problem rather than spending time trying to re-create it.

**Comment by davidfant** [comment]
@bvaughn here you go:
https://github.com/davidfant/react-resizable-panels-issue-372/blob/master/src/components/Example1.tsx

Run `yarn dev` and go to localhost:3000

**Comment by bvaughn** [comment]
Thanks. I can see the error now. Made it into a Sandbox [here](https://codesandbox.io/p/sandbox/epic-dew-6ly8t2?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clynfuwe900063b6j25ulbh4n%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clynfuwe800023b6j9su8e73j%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clynfuwe800033b6j3b81y2td%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clynfuwe800053b6jx0xoxbkb%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clynfuwe800023b6j9su8e73j%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clynfuwe800013b6juunxf7xj%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Findex.js%2522%257D%255D%252C%2522id%2522%253A%2522clynfuwe800023b6j9su8e73j%2522%252C%2522activeTabId%2522%253A%2522clynfuwe800013b6juunxf7xj%2522%257D%252C%2522clynfuwe800053b6jx0xoxbkb%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clynfuwe800043b6j00mdvqqy%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clynfuwe800053b6jx0xoxbkb%2522%252C%2522activeTabId%2522%253A%2522clynfuwe800043b6j00mdvqqy%2522%257D%252C%2522clynfuwe800033b6j3b81y2td%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clynfuwe800033b6j3b81y2td%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Comment by bvaughn** [comment]
Okay I think I see the problem here.

**Comment by bvaughn** [comment]
Fix published in version 2.0.22

---

❤️ → ☕ [givebrian.coffee](http://givebrian.coffee/)
